src/user-safe/UserSafe.sol

```diff
@@ -73,64 +47,40 @@ contract UserSafe is Initializable, OwnableUpgradeable {
     uint256 private _nonce;
     // Current spending limit
     SpendingLimitData private _spendingLimit;
-    // This spending limit gets activated once the _spendingLimit time is over
-    SpendingLimitData private _incomingSpendingLimit;
-
-    event DepositFunds(address token, uint256 amount);
-    event ApprovalFunds(address token, address spender, uint256 amount);
-    event WithdrawalRequested(
-        address[] tokens,
-        uint256[] amounts,
-        address recipient,
-        uint256 finalizeTimestamp
-    );
-    event WithdrawalProcessed(
-        address[] tokens,
-        uint256[] amounts,
-        address recipient
-    );
-    event TransferUSDCForSpending(uint256 amount);
-    event SwapTransferForSpending(uint256 weETHAmount, uint256 usdcAmount);
-    event TransferWeETHAsCollateral(uint256 amount);
-    event SetSpendingLimit(uint8 spendingLimitType, uint256 limitInUsd);
-    event SetIncomingSpendingLimit(uint8 spendingLimitType, uint256 limitInUsd);
-
-    error InsufficientBalance();
-    error ArrayLengthMismatch();
-    error CannotWithdrawYet();
-    error UnauthorizedCall();
-    error InvalidNonce();
-    error AmountGreaterThanUsdcReceived();
-    error ExceededSpendingLimit();
-
-    constructor(
-        address _usdc,
-        address _weETH,
-        address _priceProvider,
-        address _cashDataProvider,
-        address __swapper
-    ) {
-        usdc = _usdc;
-        weETH = _weETH;
-        priceProvider = IPriceProvider(_priceProvider);
+
+    constructor(address _cashDataProvider) {
         cashDataProvider = ICashDataProvider(_cashDataProvider);
```

Comment:
Feel free to remove 'CashDataProvider' thing.

But it is good to have a address registry where we can register some relevant contracts (name, address) https://github.com/etherfi-protocol/smart-contracts/blob/master/src/helpers/AddressProvider.sol

Reply:
We can use similar registry as well instead of data provider. The Cash Data Provider seems to be unnecessary but a good to have as a single source of truth. What do you suggest? Should I completely remove it?

Reply:
I don't have much preference here! your call :)

Reply:
Okay will keep it for now. If I think we don't need it, will remove.